### PR TITLE
Key phrase detected: async notifications

### PR DESCRIPTION
**Pull request by mrajwa**
DO NOT MERGE! This is just a proposal of how we can handle async notification between key phrase detector component and KPB.

Key phrase detector component sends async notification to KPB
component using notifier module. KPB registers to specific event
dedicated to key phrase detection. Once such event takes place
detection component calls notifier_event handler which in turn
calls KPB call-back function.

Signed-off-by: Marcin Rajwa <marcin.rajwa@linux.intel.com>

### CODE REVIEW

File: src/include/sof/notifier.h [hunk 1]
```diff
@@ -37,29 +37,31 @@
...
-#define NOTIFIER_ID_SSP_FREQ	1
-
 /* notifier target core masks */
 #define NOTIFIER_TARGET_CORE_MASK(x)	(1 << x)
 #define NOTIFIER_TARGET_CORE_ALL_MASK	0xFFFFFFFF
 
+enum notify_id {
+	NOTIFIER_ID_CPU_FREQ = 0,
+	NOTIFIER_ID_SSP_FREQ,
+	NOTIFIER_ID_KEY_PHRASE_DETECTED,
```
Comment (mmaka1) [inline src/include/sof/notifier.h]: Pls replace with some more generic name for all events sent by KPB clients, e.g. `NOTIFIER_ID_KPB_CLIENT_EVT'

File: src/audio/kp_detector.c [hunk 2]
```diff
@@ -0,0 +1,28 @@
+#include <sof/audio/kp_detector.h>
+#include <sof/audio/component.h>
+
+void kp_dummy_func(void);
+struct comp_data dummy_cd;
+
+/* copy and process stream data from source to sink buffers */
+static int kp_detector_copy(struct comp_dev *dev)
```
Comment (mmaka1) [inline src/audio/kp_detector.c]: Seems to be a dummy test code, a final version should be part of kp-detector commit.

File: src/include/sof/audio/kp_common.h [hunk 3]
```diff
@@ -0,0 +1,6 @@
+#include <stdint.h>
+
+struct kp_data {
```
Comment (mmaka1) [inline src/include/sof/audio/kp_common.h]: Anything that is part of kpbuf public interface (internal to other fw components/units) may go to kpb.h or kpb_public.h (the latter follows current organization where the <com>.h headers declare rather private data).

File: src/audio/kp_detector.c (hunk 2 above)
Comment (mrajwa) [reply src/audio/kp_detector.c]: Yes, I was thinking about merging this code with kp_detector once its PR is accepted.

File: src/include/sof/audio/kp_common.h (hunk 3 above)
Comment (mrajwa) [reply src/include/sof/audio/kp_common.h]: kp_common.h, or kpb_public.h as you suggested has the same intention - provide a public/common data structure to be shared by key phrase related components.

**Comment by ranj063** [comment]
@mrajwa I'm having a bit of trouble understanding the intent of this PR and the need for the kp_detector component. It seems like what you are looking for is to implement the cmd callback for the kpb component if I understand it correctly.

**Comment by mrajwa** [comment]
@ranj063 because "kp_detector" is really #1036 and here I create a dummy detector component and KPB just to show the intention. Once the #1036 and ##994 get accepted I will merge this one into them. Does it make sense now?

**Comment by ranj063** [comment]
@mrajwa why not add the commits to those PR's instead then? It will make reviewing the flow easier there. But with this PR the reviewers will have to review unnecessary code.

**Comment by mrajwa** [comment]
Well I did not want to mess with with other PRs -  especially with the detector one which was not created by myself. However I agree, it indeed might be easier to follow so I will close this PR and incorporate it into each PR accordingly.

**Pull request closed without merging**